src/core/ruleset.cpp

```diff
@@ -82,6 +72,43 @@ int Ruleset::getValue(const Card &c) {
     return 0;
 }
 
+void Ruleset::rebuildRuleInstances(){
+    m_ruleInstances.clear();
+    for(const auto pair : m_nameToRule){
+        if(isEnabled(pair.first)){
+            m_ruleInstances.push_back(pair.second);
+        }
+    }
+}
+
+void Ruleset::setEnabled(std::string rule) {
```

Comment:
maybe add a default option for diamonds, jacks, queens, h10 and Dia A?

Reply:
I think this should not be the responsibility of the ruleset class, but rather of the class that handles the players announcing soli etc.